index.js

```diff
@@ -80,7 +82,35 @@ module.exports = function (dir, opts, cb) {
       })
     }
 
-    cb(null, dat)
+    // dat.json
+    // reads to dat.meta if exists
+    // creates dat.json if not exists with title = dir name
+    // (TODO: move to module & validate dat.json)
+    var datJsonFile = path.join(dir, 'dat.json')
+    fs.readFile(datJsonFile, 'utf8', function (_, body) {
+      if (err && err.code === 'ENOENT' || !body) return createMeta()
+      else if (err) return cb(err)
+      dat.meta = JSON.parse(body)
+      done()
+    })
+
+    function createMeta () {
+      dat.meta = {title: path.basename(dir), description: ''}
+      if (dat.key) dat.meta.url = 'dat://' + datKeyAs.str(dat.key)
+      writeMeta(done)
```

Comment:
this makes the control flow a bit hard to read, what about inlining the `writeMeta` function here?